lib/unleash/feature_toggle.rb

```diff
@@ -40,8 +44,7 @@ def get_variant(context, fallback_variant = disabled_variant)
       return disabled_variant unless self.enabled && am_enabled?(context, true)
       return disabled_variant if sum_variant_defs_weights <= 0
 
-      variant = variant_from_override_match(context)
-      variant = variant_from_weights(context) if variant.nil?
+      variant = variant_from_override_match(context) || variant_from_weights(context, resolve_stickiness)
```

Comment:
I am not sure if this is a real issue or not, but if the variant is not nil, then variants from weights will be called. (Before it would not). 

But this could maybe theoretically change behavior? But I think tests would have caught it, if it would have been a real issue.

Reply:
Could be that it just wastes some CPU cycles?

Reply:
So I've run through this a million times in my mind and I've written some scratch-pad code to test this for myself. Unless there's something subtle about the way Ruby resolves function calls I don't think this is the case. 

If variant_from_override_match resolves to any truthy value then boolean short circuiting should kick in and variant_from_weights won't be called. 

If variant_from_override_match resolves to a falsey value then the  variant_from_weights will execute.

While this is technically a logic change here, since we're no longer handling just nils and now capturing that whole case with falsely values, I don't think it matters since variant_from_override_match can only return a match, a new value or a nil.

If you consider the following sample code:

class Test
  def fail_hard
    raise StandardError, 'This is broken'
  end

  def return_input(input)
    puts "I was called with #{input}"
    input
  end
end

class Empty
end

test = Test.new
test.return_input(true) || test.fail_hard
test.return_input(Empty.new) || test.fail_hard
test.return_input(nil) || test.fail_hard
`

When run this yields the following output:

I was called with true
I was called with #<Empty:0x00005593a2beda58>
I was called with 
Traceback (most recent call last):
	1: from main.rb:18:in `<main>'
main.rb:3:in `fail_hard': This is broken (StandardError)

I can't see a meaningful difference between the toy example here and the code in question.

...but it's Friday and it's been a long week so I'm open to being told I'm crazy

`

Reply:
This is a non-issue, I just wasn't able to read clearly on friday.